lib/rules/arrow-parens.js

```diff
@@ -29,18 +38,63 @@ module.exports = {
         var message = "Expected parentheses around arrow function argument.";
         var asNeededMessage = "Unexpected parentheses around single function argument";
         var asNeeded = context.options[0] === "as-needed";
-
+        var requireForBlockBodyMessage = "Unexpected parentheses around single function argument having a body with no curly braces";
+        var requireForBlockBodyNoParensMessage = "Expected parentheses around arrow function argument having a body with curly braces.";
+        var requireForBlockBody = asNeeded && context.options[1] && context.options[1].requireForBlockBody === true;
         var sourceCode = context.getSourceCode();
 
+
+
         /**
          * Determines whether a arrow function argument end with `)`
          * @param {ASTNode} node The arrow function node.
          * @returns {void}
          */
         function parens(node) {
             var token = sourceCode.getFirstToken(node);
+            var bodyToken = sourceCode.getFirstToken(node.body);
+
+            // "as-needed", { "requireForBlockBody": true }: x => x
+            if (
+                requireForBlockBody &&
+                node.params.length === 1 && node.params[0].type === "Identifier" &&
+                bodyToken.type !== "Punctuator" && bodyToken.value !== "{"
+            ) {
+                if (token.type === "Punctuator" && token.value === "(") {
+                    context.report({
+                        node: node,
+                        message: requireForBlockBodyMessage,
+                        fix: function(fixer) {
+                            var paramToken = context.getTokenAfter(token);
+                            var closingParenToken = context.getTokenAfter(paramToken);
+
+                            return fixer.replaceTextRange([
+                                token.range[0],
+                                closingParenToken.range[1]
+                            ], paramToken.value);
+                        }
+                    });
+                }
+                return;
+            }
+
+            if (
+                requireForBlockBody &&
+                bodyToken.type === "Punctuator" && bodyToken.value === "{"
```

Comment:
Same comment as above - thoughts on checking if `node.body.type` is 'BlockStatement'? Seems like a good idea to rely on info the parser is giving us, since we can.